src/services/mongoose/wallets.ts

```diff
@@ -117,7 +117,8 @@ const resultToWallet = (result: WalletRecord): Wallet => {
   const accountId = fromObjectId<AccountId>(result._accountId)
   const type = result.type as WalletType
   const currency = result.currency as WalletCurrency
-  const onChainAddressIdentifiers = result.onchain.map(({ pubkey, address }) => {
+  const onChain = result.onchain || []
```

Comment:
I'm not sure about this. Isn't it masking an underlying bug / data inconsistency.
- if this compiled before then the type must be wrong
- if the type is correct but the runtime data is wrong then we need to fix it there

Reply:
yep, it is a problem with mongoose default value... I will merge this and probably create a migration script to fix users collection (probably an issue with old users)